### DISCUSSION TITLE
Why is the compiled whl file smaller than the directly compiled .so file?

### DISCUSSION THREAD

**Discussion by sirodeneko** [PyO3/maturin#1464] [Q&A] [answered]
I found a strange thing, the whl file compiled through maturin is smaller than the .so file compiled directly. It stands to reason that because of the existence of python glue code, it will not be much smaller than the .so file. May I ask what parameters can be used to optimize the compilation size?
```
maturin
   linux : 27.8 Mb
   mac : 7.8Mb
cdylib
   linux : 40.4 Mb
   mac : 14.2Mb
```

**Comment by konstin** [comment, marked as answer]
whl files are actually zip files! you even can just rename them from `.whl` to `.zip` and look inside :)

btw if you want smaller files, try `--strip`, that's an optimization that remove dead code from the final .so
